apicapi/apic_client.py

```diff
@@ -490,6 +519,28 @@ def _save_cookie(self, request, response):
             attributes = imdata[0]['error']['attributes']
         return attributes
 
+    def _calculate_signature(self, request, url, kwargs):
+        """ Sign the request using private key and put the result in cookies"""
+        if self._is_cert_auth():
+            if url.endswith('?'):
+                url = url[:-1]
+            url = url.replace('//', '/')
+            payLoad = request + '/api' + url + kwargs.get('data', '')
+            signedDigest = sign(self.private_key, payLoad, 'sha256')
+            signature = base64.b64encode(signedDigest)
+            kwargs['cookies'] = {
+                'APIC-Request-Signature' :  signature,
+                'APIC-Certificate-Algorithm' : 'v1.0',
+                'APIC-Certificate-Fingerprint' : 'fingerprint',
+                'APIC-Certificate-DN' : self._get_cert_dn(self.username,
+                                                          self.cert_name)}
+
+    def _is_cert_auth(self):
```

Comment:
Right now, password-based authentication is given preference, so if both password and private-key is provided, we will use password. This makes maintaining the state of the boolean you are proposing a tad a complicated and error prone. So I'd prefer to calculate it on demand